nav2_amcl/src/amcl_node.cpp

```diff
@@ -1102,17 +1081,11 @@ AmclNode::freeMapDependentMemory()
     map_ = NULL;
   }
 
-  if (pf_ != NULL) {
-    pf_free(pf_);
-    pf_ = NULL;
-  }
-
-  motion_model_.reset();
-  motion_model_ = std::unique_ptr<nav2_amcl::MotionModel>(nav2_amcl::MotionModel::createMotionModel(
-        robot_model_type_, alpha1_, alpha2_, alpha3_, alpha4_, alpha5_));
-
-  // Laser
+  // Clear queued laser objects because they hold pointers to the existing
+  // map, #5202.
   lasers_.clear();
+  lasers_update_.clear();
+  frame_to_laser_.clear();
```

Comment:
I dont think these are at all related to the maps. This would be a huge pain when you have to do map sharding and you're constantly loading new section of the map in a large building. I think these should go back to where they were

Reply:
There is this comment regarding these: `// Clear queued laser objects because they hold pointers to the existing // map, #5202.` in the original code.

Reply:
So why was this in another method before?

Reply:
It was in `handleMapMessage` method right after calling the `freeMapDependentMemory`. Since the only thing that `freeMapDependentMemory` now does is to delete the map, I decided to put it in the `freeMapDependentMemory` to make the `handleMapMessage` method cleaner.